File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/BinaryAndContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import io.github.iamrenny.ruleflow.visitors.Visitor;

public class BinaryAndContextEvaluator implements ContextEvaluator<RuleFlowLanguageParser.BinaryAndContext> {

    @Override
    public Boolean evaluate(RuleFlowLanguageParser.BinaryAndContext ctx, Visitor visitor) {
        return (Boolean) visitor.visit(ctx.left) && (Boolean) visitor.visit(ctx.right);
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/BinaryAndContextEvaluator.kt
package com.github.iamrenny.rulesflow.evaluators

import com.rappi.fraud.analang.ANAParser
import com.github.iamrenny.rulesflow.visitors.Visitor

class BinaryAndContextEvaluator : ContextEvaluator<ANAParser.BinaryAndContext> {

    override fun evaluate(ctx: ANAParser.BinaryAndContext, visitor: Visitor): Boolean {
        return visitor.visit(ctx.left) as Boolean && visitor.visit(ctx.right) as Boolean
        }
}